pkg/keycloak/client.go

```diff
@@ -48,50 +50,151 @@ type TokenResponse struct {
 	ErrorDescription string `json:"error_description"`
 }
 
-func (c *Client) ListRealms() (map[string]*v1alpha1.KeycloakRealm, error) {
+func (c *Client) ListRealms() ([]*v1alpha1.KeycloakRealm, error) {
```

Comment:
Deliberately keeping the client return types consistent with what the keycloak API returns. Any manipulation of the data for reconciliation should be done outside of here.